test/parallel/test-http2-compat-expect-continue-check.js

```diff
@@ -0,0 +1,71 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const testResBody = 'other stuff!\n';
+const testResHeaderKey = 'abcd';
+const testResHeaderVal = '1';
+
+function handler(req, res) {
+  console.error('Server sent full response');
+
+  res.writeHead(200, {
+    'content-type': 'text/plain',
+    [testResHeaderKey]: [testResHeaderVal]
+  });
+  res.end(testResBody);
+}
+
+const server = http2.createServer(
+  common.mustNotCall('Full response sent before 100 Continue')
+);
+
+server.on('checkContinue', common.mustCall((req, res) => {
+  console.error('Server received Expect: 100-continue');
+
+  res.writeContinue();
+
+  setTimeout(common.mustCall(() => handler(req, res)), 100);
+}));
+
+server.listen(0);
+
+server.on('listening', common.mustCall(() => {
+  let body = '';
+
+  const port = server.address().port;
+  const client = http2.connect(`http://localhost:${port}`);
+  const req = client.request({
+    ':method': 'POST',
+    ':path': '/world',
+    expect: '100-continue'
+  });
+  console.error('Client sent request');
+
+  req.on('continue', common.mustCall(() => {
+    console.error('Client received 100-continue');
```

Comment:
we need to verify that this event happens before the response

Reply:
Ah, yeah. Good point. Lost that when I refactored these tests. Will fix.